packages/flutter_tools/lib/src/compile.dart

```diff
@@ -540,11 +547,12 @@ class DefaultResidentCompiler implements ResidentCompiler {
     this.platformDill,
     List<String> dartDefines,
     this.librariesSpec,
+    @visibleForTesting StdoutHandler stdoutHandler,
```

Comment:
Why pass an instance here, but a factory above? Can both do the same thing?

Reply:
Because the kernel compiler creates a new instance every time you tell it to compile, while the resident re-uses one. I think I could change the logic so that kernel compiler also sets up one since we don't actually re-use it.

Reply:
Made this behave the same as the ResidentCompiler